### PR TITLE
cleanup: Use coalesce expression

### PR DESCRIPTION
**Pull request by xtqqczze**
# PR Summary

Automate fix of [RCS1128](https://github.com/JosefPihrt/Roslynator/blob/master/docs/analyzers/RCS1128.md)

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/6/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [x] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [x] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and enumerated concerns in the summary. This may include:
        - Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode (which runs in a different PS Host).
        - Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
        - Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
        - Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).

### CODE REVIEW

File: src/System.Management.Automation/engine/PSConfiguration.cs [hunk 1]
```diff
@@ -219,38 +219,25 @@ internal void SetExperimentalFeatures(ConfigScope scope, string featureName, boo
...
-            if (settingValue == null)
-            {
-                // if the setting is not mentioned in configuration files, then the default WindowsPowerShellCompatibilityModuleDenyList value is null
-                settingValue = ReadValueFromFile<string[]>(ConfigScope.AllUsers, WindowsPowerShellCompatibilityModuleDenyListKey);
-            }
-
-            return settingValue;
+            return ReadValueFromFile<string[]>(ConfigScope.CurrentUser, WindowsPowerShellCompatibilityModuleDenyListKey)
+                ?? ReadValueFromFile<string[]>(ConfigScope.AllUsers, WindowsPowerShellCompatibilityModuleDenyListKey)
+                ?? null;
```
Comment (iSazonov) [inline src/System.Management.Automation/engine/PSConfiguration.cs:233]: Why not:
```suggestion
                ?? ReadValueFromFile<string[]>(ConfigScope.AllUsers, WindowsPowerShellCompatibilityModuleDenyListKey);
```

File: src/System.Management.Automation/engine/PSConfiguration.cs [hunk 2]
```diff
@@ -219,38 +219,25 @@ internal void SetExperimentalFeatures(ConfigScope scope, string featureName, boo
...
-            if (settingValue == null)
-            {
-                // if the setting is not mentioned in configuration files, then the default WindowsPowerShellCompatibilityNoClobberModuleList value is null
-                settingValue = ReadValueFromFile<string[]>(ConfigScope.AllUsers, WindowsPowerShellCompatibilityNoClobberModuleListKey);
-            }
-
-            return settingValue;
+            return ReadValueFromFile<string[]>(ConfigScope.CurrentUser, WindowsPowerShellCompatibilityNoClobberModuleListKey)
+                ?? ReadValueFromFile<string[]>(ConfigScope.AllUsers, WindowsPowerShellCompatibilityNoClobberModuleListKey)
+                ?? null;
```
Comment (iSazonov) [inline src/System.Management.Automation/engine/PSConfiguration.cs:240]:
```suggestion
                ?? ReadValueFromFile<string[]>(ConfigScope.AllUsers, WindowsPowerShellCompatibilityNoClobberModuleListKey);
```

File: src/System.Management.Automation/engine/PSConfiguration.cs (hunk 1 above)
Comment (xtqqczze) [reply src/System.Management.Automation/engine/PSConfiguration.cs:233]: I wanted to state the default value explicitly, so the code could be self-documenting.

Comment (iSazonov) [reply src/System.Management.Automation/engine/PSConfiguration.cs:233]: I think it makes no sense in the case.

**Review by iSazonov — APPROVED** [review]

File: src/System.Management.Automation/engine/PSConfiguration.cs [hunk 3]
```diff
@@ -219,38 +219,23 @@ internal void SetExperimentalFeatures(ConfigScope scope, string featureName, boo
...
-            {
-                // if the setting is not mentioned in configuration files, then the default DisableImplicitWinCompat value is False
-                settingValue = ReadValueFromFile<bool?>(ConfigScope.AllUsers, DisableImplicitWinCompatKey, defaultValue: false);
-            }
+            bool settingValue = ReadValueFromFile<bool?>(ConfigScope.CurrentUser, DisableImplicitWinCompatKey)
+                ?? ReadValueFromFile<bool?>(ConfigScope.AllUsers, DisableImplicitWinCompatKey)
+                ?? false;
 
-            return !settingValue.Value;
+            return !settingValue;
```
Comment (TravisEz13) [inline src/System.Management.Automation/engine/PSConfiguration.cs:226]: is this going to give the same result

File: src/System.Management.Automation/security/wldpNativeMethods.cs [hunk 4]
```diff
@@ -82,7 +82,7 @@ public static SystemEnforcementMode GetLockdownPolicy(string path, SafeHandle ha
             // At this point, LockdownPolicy = Audit or Allowed.
             // If there was a WLDP policy, but WLDP didn't block it,
             // then it was explicitly allowed. Therefore, return the result for the file.
-            SystemEnforcementMode systemWldpPolicy = s_cachedWldpSystemPolicy.GetValueOrDefault(SystemEnforcementMode.None);
+            SystemEnforcementMode systemWldpPolicy = s_cachedWldpSystemPolicy ?? SystemEnforcementMode.None;
```
Comment (TravisEz13) [inline src/System.Management.Automation/security/wldpNativeMethods.cs:85]: Can you revert the changes to this file and make them in a separate PR?  They will require more review.

**Review by TravisEz13 — CHANGES_REQUESTED** [review]
see comments in wldpNativeMethods

File: src/System.Management.Automation/engine/PSConfiguration.cs (hunk 3 above)
Comment (iSazonov) [reply src/System.Management.Automation/engine/PSConfiguration.cs:226]: The type is changed above. Looks good.

**Comment by TravisEz13** [comment]
@iSazonov Please don't put the milestone until PRs are merged.

**Review by TravisEz13 — APPROVED** [review]
